### PR TITLE
CA1062: rule now support api_surface

### PR DESCRIPTION
**Pull request by Evangelink**
Relates to [#5881](https://github.com/dotnet/roslyn-analyzers/pull/5881).

Do not merge before linked PR is merged.

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca1062.md [hunk 1]
```diff
@@ -48,13 +48,16 @@ You can suppress a warning from this rule if you are sure that the dereferenced
...
 
+- [Include specific API surfaces](#include-specific-api-surfaces)
 - [Exclude specific symbols](#exclude-specific-symbols)
 - [Exclude specific types and their derived types](#exclude-specific-types-and-their-derived-types)
 - [Exclude extension method 'this' parameter](#exclude-extension-method-this-parameter)
 - [Null check validation methods](#null-check-validation-methods)
 
 These options can be configured for just this rule, for all rules, or for all rules in this category ([Design](design-warnings.md)). For more information, see [Code quality rule configuration options](../code-quality-rule-options.md).
 
+[!INCLUDE[api-surface](~/includes/code-analysis/api-surface.md)]
```
Comment (mavasani) [inline docs/fundamentals/code-analysis/quality-rules/ca1062.md:59]: Do we need to specify that the option is supported only in .NET7 and later?

Comment (Evangelink) [reply docs/fundamentals/code-analysis/quality-rules/ca1062.md:59]: How do we handle such change generally? Like every new user option is only available on new version of the code, right? For example, do we have the same concern with #28234?

Comment (gewarren) [reply docs/fundamentals/code-analysis/quality-rules/ca1062.md:59]: We should note the version somehow. We should also add CA1062 to this list: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-quality-rule-options#api_surface.

```suggestion
[!INCLUDE[api-surface](~/includes/code-analysis/api-surface.md)]

> [!NOTE]
> This option is supported for CA1062 in .NET 7 and later versions only.
```

File: docs/fundamentals/code-analysis/code-quality-rule-options.md [hunk 2]
```diff
@@ -80,7 +80,7 @@ This section lists some of the available options. To see the full list of availa
...
....md) [CA1028](quality-rules/ca1028.md)<br/>[CA1030](quality-rules/ca1030.md) [CA1036](quality-rules/ca1036.md) [CA1040](quality-rules/ca1040.md) [CA1041](quality-rules/ca1041.md)<br/>[CA1043](quality-rules/ca1043.md) [CA1044](quality-rules/ca1044.md) [CA1051](quality-rules/ca1051.md) [CA1052](quality-rules/ca1052.md)<br/>[CA1054](quality-rules/ca1054.md) [CA1055](quality-rules/ca1055.md) [CA1056](quality-rules/ca1056.md) [CA1058](quality-rules/ca1058.md)<br/>[CA1062](quality-rules/ca1062.md) [CA1063](quality-rules/ca1063.md) [CA1708](quality-rules/ca1708.md) [CA1710](quality-rules/ca1710.md)<br/>[CA1711](quality-rules/ca1711.md)[CA1714](quality-rules/ca1714.md) [CA1715](quality-rules/ca1715.md) [CA1716](quality-rules/ca1716.md)<br/>[CA1717](quality-rules/ca1717.md)[CA1720](quality-rules/ca1720.md) [CA1721](quality-rules/ca1721.md) [CA1725](quality-rules/ca1725.md)<br/>[CA1801](quality-rules/ca1801.md) [CA1802](quality-rules/ca1802.md) [CA1815](quality-rules/ca1815.md) [CA1819](quality-rules/ca1819.md)<br/>[CA2217](quality-rules/ca2217.md) [CA2225](quality-rules/ca2225.md) [CA2226](quality-rules/ca2226.md) [CA2231](quality-rules/ca2231.md)<br/>[CA2234](quality-rules/ca2234.md)<br/>|
```
Comment (Evangelink) [inline docs/fundamentals/code-analysis/code-quality-rule-options.md:83]: The addition of the `<br/>` makes it complex to update the list.

**Review by gewarren — APPROVED** [review]

**Pull request merged**